src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1482,7 +1482,13 @@ LedgerManagerImpl::transferLedgerEntriesToBucketList(
         {
             LedgerTxn ltxEvictions(ltx);
             mApp.getBucketManager().scanForEviction(ltxEvictions, ledgerSeq);
-            if (ledgerCloseMeta)
+
+            // See comment below in LedgerManagerImpl::ledgerClosed for info
+            // about the ledgerCloseMeta protocol version check.
+            if (ledgerCloseMeta &&
+                protocolVersionStartsFrom(
+                    ledgerCloseMeta->getInitialProtocolVersion(),
+                    ProtocolVersion::V_20))
```

Comment:
I think scanForEviction should be behind the protocol-used-to-apply check as well. [This comment ](https://github.com/stellar/stellar-core/issues/3909#issuecomment-1709338073)suggests passing the old protocol version to `ledgerClosed`